cheri/cheri_insts.sail

```diff
@@ -119,6 +119,23 @@ function clause execute (CGetAndAddr(rd, cb, rs)) =
   wGPR(rd) = capVal.address & rs_val;
 }
 
+union clause ast = CHasSpace : (regno, regno, regno)
+function clause execute (CHasSpace (rd, cb, rs)) =
+{
+  checkCP2usable();
+  let cb_val = readCapReg(cb);
+  let cb_cursor = getCapCursor(cb_val);
+  let cb_base = getCapBase(cb_val);
+  let cb_top = getCapTop(cb_val);
+  let rs_val = unsigned(rGPR(rs));
+  if ((cb_cursor + rs_val) <= cb_top) &
+     (cb_cursor < cb_top) &
+     (cb_cursor >= cb_base) then
+    wGPR(rd) = zero_extend(to_bits(1,1))
```

Comment:
```suggestion
    wGPR(rd) = zero_extend(0b1)
```

And maybe do something like either CTestSubset or CGetType/CToPtr to avoid multiple `wGPR(rd) = ` statements.